rfc/2023/2023-05-01-va-rfc-frontend-performance.md

```diff
@@ -0,0 +1,156 @@
+
+Datadog and Google Lighthouse help us understand the Veteran experience with VA.gov by measuring frontend performance. The information provided can help teams identify how to improve website performance for Veterans using various devices and internet connection speeds. These measurements can also help teams define and adopt a budget for how much HTML, CSS and JavaScript should be served per application and per pageview.
+
+We can address many challenges to frontend performance on VA.gov today. Every performance report on the [Frontend Performance Dashboard](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/frontend-support-dashboard/lighthouse-performance-report/) links to a "Treemap", a visualization of what JavaScript libraries and components are included and estimates for how much is unused or duplicated. For example, the treemap for the [Va.gov homepage's report](https://vetsgov-website-builds-s3-upload-test.s3-us-gov-west-1.amazonaws.com/lighthouse/homepage.html) shows that of 3.59MB of JavaScript, 1.55MB, or about 43% is *unused*.
+
+Further digging into existing performance reports, it appears that most, if not all, pages have three JavaScript bundles, and each bundle has its own copy of React. It appears that **pages have multiple redundant copies of the same code libraries**. In part, this is due to pages including `web-components.entry.js` (508.752 KiB) and `vendor.entry.js` (349.064 KiB) in addition to an application-specific bundle. The tools we use today can [prevent this duplication](https://webpack.js.org/guides/code-splitting/#prevent-duplication).
+
+A sample of performance scores is available below under [Performance Scores](#performance-scores)
+
+## Design
+
+> Explain the proposed design in enough detail so that a team member will fully understand the implementation. Include a diagram (in the `images` dir) as needed to convey your plans. Use active voice, present tense, and decisive language.
+
+The Platform should provide general guidance and technical resources to empower VFS teams to adopt performance budgets. Providing access to Datadog's RUM is a first step that will give teams information about performance via Core Web Vitals, which can help them come up with performance budget targets. Teams are already in the process of gaining access to Datadog, and adding RUM to measure Core Web Vitals is a reasonable next step.
+
+Sitewide performance issues can be addressed to improve performance across VA.gov applications. The **shared header and footer** are React applications that are part of the `static-pages.entry.js` bundle, which is 1.397MB as of April 2023. The header and footer should be refactored to remove React and use build-time generated HTML, with lightweight web components to add interactivity and manage accessibility. Improving performance of the header and footer will have a positive impact across VA.gov and subdomains that inject the header and footer.
```

Comment:
Probably mention teamsites there just to set the tone on the LOE for this since it's a little beyond just rebuilding the header/footer

Reply:
Is there something to say more specifically than "rebuilding the header/footer will require some teamsites work"? I'm not familiar with the implementation for teamsites…

Reply:
It's basically just an async loaded header an footer into HTML we can't change or accurately test before release. I think mentioning it is enough for this instance